godot-core/src/builtin/string/node_path.rs

```diff
@@ -37,8 +43,33 @@ impl NodePath {
         Self { opaque }
     }
 
-    pub fn is_empty(&self) -> bool {
-        self.as_inner().is_empty()
+    /// Returns the number of node names in the path. Property subnames are not included.
+    pub fn get_name_count(&self) -> u32 {
+        self.as_inner()
+            .get_name_count()
+            .try_into()
+            .expect("Godot name counts are non negative int")
+    }
+
+    /// Returns the node name indicated by `idx`, starting from 0. If `idx` is out of bounds, [`None`] is returned.
+    /// See also [`NodePath::get_subname_count`] and [`NodePath::get_name_count`].
+    pub fn get_name(&self, idx: u32) -> Option<StringName> {
```

Comment:
Function docs should ideally have a "brief" description that's only 1 line (typically 1 sentence), and then a body regarding all the edge cases and details. The brief should be as concise as possible, because it's the first thing a user reads -- so the user can quickly decide whether that function is what they need.

We also typically format method names as 
```
[`method()`][Self::method]
```
That way, you have both a short term visible in text, but also a link.

An example would also be nice. So this could be:
```suggestion
    /// Returns the node name at position `idx`.
    ///
    /// If `idx` >= [`get_name_count()`][Self::get_name_count], `None` is returned.
    ///
    /// If you want to include properties, check out [`get_subname()`][Self::get_subname].
    /// 
    /// # Example
    /// ```no_run
    /// # use godot::prelude::*;
    /// let path = NodePath::from("../RigidBody2D/Sprite2D");
    /// godot_print!("{}", path.get_name(0).unwrap()); // ".."
    /// godot_print!("{}", path.get_name(1).unwrap()); // "RigidBody2D"
    /// godot_print!("{}", path.get_name(2).unwrap()); // "Sprite"
    /// ```
    pub fn get_name(&self, idx: u32) -> Option<StringName> {
```
You should be able to directly commit the suggestion :slightly_smiling_face:

Reply:
I adapted the Godot docs to make sense as Rust docs, so notmyfault™. That said, I'd change the following:

```rust
If you want to get section of the property path instead, check out [`get_subname()`][Self::get_subname].
```

Since, to my knowledge, `get_subname` only returns parts of the property path, not of the node path part.

Reply:
Good point, your wording is indeed better!